pkg/provider/garden/ruleset/securityhardenedshoot/rules/2007.go

```diff
@@ -0,0 +1,137 @@
+
+import (
+	"context"
+	"slices"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubectl/pkg/scheme"
+	admissionapiv1 "k8s.io/pod-security-admission/admission/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/diki/pkg/rule"
+	"github.com/gardener/diki/pkg/shared/ruleset/disak8sstig/option"
+)
+
+var (
+	_ rule.Rule     = &Rule2007{}
+	_ rule.Severity = &Rule2007{}
+	_ option.Option = &Options2007{}
+)
+
+type Options2007 struct {
+	MinPodSecurityLevel string `json:"minPodSecurityLevel" yaml:"minPodSecurityLevel"`
+}
+
+func (o Options2007) Validate() field.ErrorList {
+	if !slices.Contains([]string{"restricted", "baseline", "privileged"}, o.MinPodSecurityLevel) && len(o.MinPodSecurityLevel) > 0 {
+		return field.ErrorList{field.Invalid(field.NewPath("minPodSecurityLevel"), o.MinPodSecurityLevel, "must be one of 'restricted', 'baseline' or 'privileged'")}
+	}
+	return nil
+}
+
+type Rule2007 struct {
+	Client         client.Client
+	Options        *Options2007
+	ShootName      string
+	ShootNamespace string
+}
+
+func (r *Rule2007) ID() string {
+	return "2007"
+}
+
+func (r *Rule2007) Name() string {
+	return "Shoot clusters must have a PodSecurity admission plugin configured."
+}
+
+func (r *Rule2007) Severity() rule.SeverityLevel {
+	return rule.SeverityHigh
+}
+
+func (r *Rule2007) Run(ctx context.Context) (rule.RuleResult, error) {
+	shoot := &gardencorev1beta1.Shoot{ObjectMeta: metav1.ObjectMeta{Name: r.ShootName, Namespace: r.ShootNamespace}}
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(shoot), shoot); err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), rule.NewTarget("name", r.ShootName, "namespace", r.ShootNamespace, "kind", "Shoot"))), nil
+	}
+
+	var podSecurityPlugin = "PodSecurity"
+
+	if shoot.Spec.Kubernetes.KubeAPIServer == nil || shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins == nil {
+		return rule.Result(r, rule.FailedCheckResult("PodSecurity admission plugin is not configured.", rule.NewTarget())), nil
+	}
+
+	if slices.ContainsFunc(shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins, func(admissionPlugin gardencorev1beta1.AdmissionPlugin) bool {
+		return admissionPlugin.Name == podSecurityPlugin && admissionPlugin.Disabled != nil && *admissionPlugin.Disabled
+	}) {
+		return rule.Result(r, rule.FailedCheckResult("PodSecurity admission plugin is disabled.", rule.NewTarget())), nil
+	}
+
+	enabledPodSecurityPluginIdx := slices.IndexFunc(shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins, func(admissionPlugin gardencorev1beta1.AdmissionPlugin) bool {
+		return admissionPlugin.Name == podSecurityPlugin
+	})
+
+	if enabledPodSecurityPluginIdx < 0 {
+		return rule.Result(r, rule.FailedCheckResult("PodSecurity admission plugin is not configured.", rule.NewTarget())), nil
+	}
```

Comment:
```suggestion
	if enabledPodSecurityPluginIdx < 0 || shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins[enabledPodSecurityPluginIdx].Config == nil {
		return rule.Result(r, rule.FailedCheckResult("PodSecurity admission plugin has a default security configuration.", rule.NewTarget())), nil
	}
```

These 2 cases should have the same message, since in both cases the `PodSecurity` plugin had been defaulted and enabled. ref https://github.com/gardener/gardener/blob/0b2f2a117b13f2c4e37bfd083e9a5b511e340071/pkg/utils/validation/admissionplugins/admissionplugins.go#L61